deps/childprocess.lua

```diff
@@ -164,4 +165,144 @@ local function spawn(command, args, options)
   return em
 end
 
+---- Exec and execfile
+
+local function normalizeExecArgs(command, arguments)
+  local file, args
+
+  local options = arguments.options or nil
+  local callback = arguments.callback
+
+  local isWindows
+  if not pcall(function() isWindows = require('luvipath').isWindows end) then
+    isWindows = los.type() == 'win32'
+  end
+
+  local stringArgs = command..table.concat(arguments, ' ')
+  if isWindows then
+    file = 'cmd.exe'
+    args = {'/s', '/c', '"'..stringArgs..'"' }
+  else
+    file = '/bin/sh'
+    args = {'-c', stringArgs }
+  end
+
+  if options and options.shell then file = options.shell end
+
+  return {
```

Comment:
Just a style nit, but I prefer multiple return values here over creating a temporary table that will simple be destructured on the other side.  Also it seems you're never using the `cmd` property returned here and it's just one of the input parameters anyway, the caller already knows it.